src/controllers/register/register.controller.ts

```diff
@@ -1,14 +1,20 @@
-import { Controller, Post, Req } from '@nestjs/common';
+import {
+  Controller,
+  Post,
+  Body,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
 import { RegisterService } from 'src/services/register/register.service';
-import { Request } from 'express';
+import { AddUserDto } from 'src/dtos/user.dto';
 
 @Controller('register')
 export class RegisterController {
   constructor(private readonly registerService: RegisterService) {}
 
   @Post()
-  addUser(@Req() request: Request): void {
-    const user = request.body;
-    return this.registerService.addUser(user);
+  @UsePipes(ValidationPipe)
```

Comment:
Could you describe your controller with OpenApi spec? 
In Nest.js is a cool integration with the Swagger -> https://docs.nestjs.com/openapi/introduction